src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java

```diff
@@ -887,6 +887,8 @@ public void remove() {
                     isParameter = true;
                     isDefaultValue = ((CredentialsParameterValue) parameter).isDefaultValue();
                     id = ((CredentialsParameterValue) parameter).getValue();
+                    // Avoid spotbugs complaining about the id being null and used lately on CredentialsMatchers.withId
```

Comment:
Is this fixing a real bug, or do we just want to `@SuppressFBWarnings`?

Reply:
AFAICT `getValue` can return a `null` and the constructor of a `CredentialsParameterValue` allows you to pass a `null`. Maybe the current use doesn't imply a bug, but a future use in tests or whatever could land on a real bug.

Reply:
So the initial credentials api did not mandate an ID because the idea was to use it as a marker interface. Very quickly we discovered that wasn’t great because everyone had to implement their own select UI... so then we added IdCredentials to enforce that.

In theory, someone could have a plugin depending on the 2012-era api and not having an ID... those would be `null` but of limited utility as you cannot integrate them widely.

At some point we should declare a major version bump and move ID to the main interface and deprecating IdCredentials. I do not think today is that day 😂

Reply:
Oh I'd love to be onboard with a version 2 plugin sometime. We can remove all the deprecated stuff, merge interface extensions back together, and perhaps make other API improvements.